### ISSUE TITLE
jitsi-web: cron unreasonable resource usage

### ISSUE THREAD

**Issue by pushytoxin** [spantaleev/matrix-docker-ansible-deploy#1028]
The Jitsi-Web docker image hosts a cron that's supposed to update the Let's Encrypt certificates. Even without configured certificates for this container, there's still a constant 10-20kB/s IO write coming from this process.

`docker stats` output:
```
CONTAINER ID   NAME                      CPU %     MEM USAGE / LIMIT    MEM %     NET I/O           BLOCK I/O         PIDS                                                [0/1893]
0f3b8edb1835   matrix-dimension          0.00%     25.52MiB / 1.89GiB   1.32%     15.4MB / 38.1MB   345MB / 54.8MB    11
70e706409700   matrix-mautrix-facebook   0.00%     2.484MiB / 1.89GiB   0.13%     24.8kB / 11.2kB   38.2MB / 0B       4
1e64c148e54b   matrix-nginx-proxy        1.40%     10.73MiB / 1.89GiB   0.55%     4.19GB / 4.72GB   57.3MB / 0B       2
af838f7d53b5   matrix-etherpad           0.29%     10.7MiB / 1.89GiB    0.55%     1.6MB / 16.2kB    183MB / 16.4kB    13
c9b5e4d5c202   matrix-synapse            24.62%    483.6MiB / 1.89GiB   24.98%    5.28GB / 14.6GB   981MB / 0B        23
c1e437c0994e   matrix-ma1sd              0.17%     3.422MiB / 1.89GiB   0.18%     37.8kB / 31.2kB   156MB / 0B        36
61ebda68a47c   matrix-jitsi-jvb          0.37%     30.16MiB / 1.89GiB   1.56%     111MB / 104MB     244MB / 20.5kB    45
0434d425c3e8   matrix-jitsi-jicofo       0.34%     52.08MiB / 1.89GiB   2.69%     110MB / 12.9MB    371MB / 20.5kB    280
eb252f32037d   matrix-jitsi-prosody      0.00%     18.27MiB / 1.89GiB   0.94%     117MB / 221MB     30.7MB / 295kB    10
62f6945710e1   matrix-jitsi-web          0.81%     8.336MiB / 1.89GiB   0.43%     29.2kB / 4.64MB   119MB / 2.19GB    9
99e693382c7f   matrix-grafana            6.38%     39.45MiB / 1.89GiB   2.04%     5.19MB / 3.82MB   520MB / 4.68MB    11
1ed0b5a2941e   matrix-prometheus         0.63%     106.4MiB / 1.89GiB   5.50%     8.9GB / 34.2MB    1.87GB / 401MB    9
3df2089321f5   matrix-synapse-admin      0.00%     1.258MiB / 1.89GiB   0.06%     13.5kB / 0B       4.78MB / 8.19kB   3
a70632570ec8   matrix-coturn             0.53%     3.574MiB / 1.89GiB   0.18%     15.9MB / 26.6MB   39.1MB / 0B       7
ae496ce7e39f   matrix-client-element     0.00%     1.172MiB / 1.89GiB   0.06%     164kB / 14.8MB    32.5MB / 0B       2
57d5a24f8155   matrix-postgres           2.82%     184.5MiB / 1.89GiB   9.53%     2.24GB / 3.24GB   4.58GB / 6.51GB   22
7acffe42b620   matrix-mailer             0.00%     6.492MiB / 1.89GiB   0.34%     46.6kB / 26.9kB   111MB / 0B        2
```

It writes 1/3 of what the database does on a small homeserver.

`fatrace` show the `s6-supervise` process (command line: `s6-supervise cron`) writing the following files:

```
s6-supervise(3255309): + /var/lib/docker/overlay2/d3f6d620de15826356637dc8971ecfc7b9567f2b3a8986d79f14aad06c5ddd04/diff/run/s6/services/cron/supervise
s6-supervise(3255309): <> /var/lib/docker/overlay2/d3f6d620de15826356637dc8971ecfc7b9567f2b3a8986d79f14aad06c5ddd04/diff/run/s6/services/cron/supervise
s6-supervise(3255309): + /var/lib/docker/overlay2/d3f6d620de15826356637dc8971ecfc7b9567f2b3a8986d79f14aad06c5ddd04/diff/run/s6/services/cron/supervise
s6-supervise(3255309): <> /var/lib/docker/overlay2/d3f6d620de15826356637dc8971ecfc7b9567f2b3a8986d79f14aad06c5ddd04/diff/run/s6/services/cron/supervise
s6-supervise(3255309): RC /var/lib/docker/overlay2/d3f6d620de15826356637dc8971ecfc7b9567f2b3a8986d79f14aad06c5ddd04/diff/run/s6/services/cron/event
s6-supervise(3255309): + /var/lib/docker/overlay2/d3f6d620de15826356637dc8971ecfc7b9567f2b3a8986d79f14aad06c5ddd04/diff/run/s6/services/cron/supervise
s6-supervise(3255309): <> /var/lib/docker/overlay2/d3f6d620de15826356637dc8971ecfc7b9567f2b3a8986d79f14aad06c5ddd04/diff/run/s6/services/cron/supervise
s6-supervise(3255309): R /var/lib/docker/overlay2/d3f6d620de15826356637dc8971ecfc7b9567f2b3a8986d79f14aad06c5ddd04/diff/run/s6/services/cron/event
s6-supervise(3255309): C /var/lib/docker/overlay2/d3f6d620de15826356637dc8971ecfc7b9567f2b3a8986d79f14aad06c5ddd04/diff/run/s6/services/cron/event
```

Apparently, this container doesn't have the `/run/` directory mounted as tmpfs.

I'm posting this issue for visibility, even though I'm not sure it's our job to fix this. What's the best course of action?

- fix the jitsi-web Dockerfile?
- disable cron?
- pull the plug on the jitsi-web container and add its configuration to matrix-nginx-proxy, as the functionality seems mostly redundant anyway?

**Comment by pushytoxin** [comment]
The service orchestrator keeps trying to start `cron`, but this is the script it runs:

```bash
#!/usr/bin/with-contenv bash

if [[ $DISABLE_HTTPS -ne 1 ]]; then
    if [[ $ENABLE_LETSENCRYPT -eq 1 ]]; then
      exec cron -f
    fi
fi
```

So it runs in a loop forever.

[This was fixed upstream](https://github.com/jitsi/docker-jitsi-meet/commit/ba011900fb17933e198a8cc4ceb23e73bb49f9e8), merged in stable-5142-1.  **We _barely_ got this fix :smile:** 

If all other issues holding back upgrading Jitsi are resolved, this problem will go away too.

**Comment by pushytoxin** [comment]
Please, github, link PR #1037

**Comment by pushytoxin** [comment]
Closed by #1037

**Comment by spantaleev** [comment]
Should be fixed by #1037, thanks to @pushytoxin!
